src/Models/CreditCard.php

```diff
@@ -163,26 +164,23 @@ private function validateInput($values)
             throw new InvalidArgumentException("The country of the address is missing.");
         }
 
-        $card = CreditCardValidator::validCreditCard($values['number']);
-        if ($card['valid'] !== true)
-        {
+        $cardValidation = CreditCardValidator::validCreditCard($values['number']);
+        if ($cardValidation['valid'] !== true) {
             throw new InvalidArgumentException("The credit card number is not valid.");
         }
 
-        if (CreditCardValidator::validCvc($values['cvc'], $card['type']) === false)
-        {
-            throw new InvalidArgumentException("The CVC is not valid.");
-        }
+        $cvv2 = $values['cvc'];
+        $cvv2 = $cardValidation['type'] === CreditCardType::AMEX ?
+            sprintf('%04d', $cvv2) :
+            sprintf('%03d', $cvv2);
```

Comment:
Nit: This could be generalized to use the value in the cards array assuming AMEX is only 4 which I am pretty certain of.